src/bin/signal_handler.rs

```diff
@@ -69,9 +69,11 @@ mod imp {
                 SIGUSR1 => {
                     // Use SIGUSR1 to log metrics.
                     let mut buffer = vec![];
-                    let metric_familys = prometheus::gather();
                     let encoder = TextEncoder::new();
-                    encoder.encode(&metric_familys, &mut buffer).unwrap();
+                    let metric_familys = prometheus::gather();
+                    for mf in metric_familys {
+                        let _ = encoder.encode(&[mf], &mut buffer);
```

Comment:
Some errors[0] will occur if there is not txn read or txn write request (eg., `tikv_scheduler_kv_command_key_read`[1]) since start. I think these errors are ignorable[2], and I don't want tikv panic when we send `SIGUSR1`.

[0] https://github.com/pingcap/rust-prometheus/pull/131/files#diff-242166b43d4a390f7d86a9d281a68a74R39
[1] https://github.com/pingcap/tikv/blob/3ebb9cf637758be5b47412432ab7c8e3e6f0f8bb/src/storage/metrics.rs#L89
[2] they are also ignored in golang's push client. https://github.com/prometheus/client_golang/blob/06bc6e01f4baf4ee783ffcd23abfcb0b0f9dfada/prometheus/push/push.go#L118

Reply:
Is it correct to return error? The situation looks OK and should not produce error. And what about other error? Should all the errors be ignored?

Reply:
Any suggestions?

Reply:
It depends on what can cause the problem. It would be better if Prometheus skips the empty metrics. In worse case, TiKV should warn about the error at lease.